test/util/client/azure/compute.go

```diff
@@ -0,0 +1,356 @@
+)
+
+// ResourceGroupFromManagedResourceID returns the resource group name from  a managed resource group identifier. The managed resource
+// group id for managed applications is formatted as follows `/subscriptions/(.+)/resourceGroups/(.+)`
+func (az *Client) ResourceGroupFromManagedResourceID(resourceID string) (string, error) {
+	const resourceIDPatternText = `(?i)subscriptions/(.+)/resourceGroups/(.+)`
+	resourceIDPattern := regexp.MustCompile(resourceIDPatternText)
+	match := resourceIDPattern.FindStringSubmatch(resourceID)
+	if len(match) != 3 {
+		return "", fmt.Errorf("parsing failed for %s. Invalid managed resource group Id format", resourceID)
+	}
+	return match[2], nil
+}
+
+// ApplicationResourceGroup returns the name of the resource group holding an OpenShift on Azure application
+// instance. This resource group may only contain one resource
+func (az *Client) ApplicationResourceGroup() string {
+	return strings.Join([]string{"OS", az.resourceGroup, az.resourceGroup, az.location}, "_")
+}
+
+// ManagedResourceGroup returns the name of the resource group holding all resources required by an OpenShift on Azure
+// managed application instance
+func (az *Client) ManagedResourceGroup(applicationResourceGroup string) (string, error) {
+	apps, err := az.appsc.ListByResourceGroup(az.ctx, applicationResourceGroup)
+	if err != nil {
+		return "", err
+	}
+
+	for _, app := range apps.Values() {
+		if appid := *app.ManagedResourceGroupID; appid != "" {
+			r, err := az.ResourceGroupFromManagedResourceID(appid)
+			if err != nil {
+				return "", err
+			}
+			return r, nil
+		}
+	}
+	return "", nil
+}
+
+// ScaleSets returns a slice of VirtualMachineScaleSets within a given resource group
+func (az *Client) ScaleSets(resourceGroup string) ([]compute.VirtualMachineScaleSet, error) {
+	vmssPages, err := az.ssc.List(az.ctx, resourceGroup)
+	if err != nil {
+		return nil, err
+	}
+
+	var scaleSets []compute.VirtualMachineScaleSet
+	for vmssPages.NotDone() {
+		scaleSets = append(scaleSets, vmssPages.Values()...)
+		err = vmssPages.Next()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return scaleSets, nil
+}
+
+// ScaleSetVMs returns a slice of VirtualMachineScaleSetVMs within a given scale set
+func (az *Client) ScaleSetVMs(resourceGroup string, scaleSet string) ([]compute.VirtualMachineScaleSetVM, error) {
+	vmPages, err := az.ssvmc.List(az.ctx, resourceGroup, scaleSet, "", "", "")
+	if err != nil {
+		return nil, err
+	}
+
+	var vms []compute.VirtualMachineScaleSetVM
+	for vmPages.NotDone() {
+		vms = append(vms, vmPages.Values()...)
+		err = vmPages.Next()
+		if err != nil {
+			return nil, err
+		}
+	}
+	return vms, nil
+}
+
+// ScaleSet returns a VirtualMachineScaleSet with a given name within a resource group
+func (az *Client) ScaleSet(resourceGroup, scaleSetName string) (*compute.VirtualMachineScaleSet, error) {
```

Comment:
Functions wrapping a single function call are probably unnecessary. Inline this Get wherever ScaleSet is used.